tests/html/shim.html

```diff
@@ -16,102 +16,299 @@
+    ES5ReflectReflect2.prototype.constructor = ES5ReflectReflect2;
+
+    customElements.define('es5-reflect-element-2', ES5ReflectReflect2);
+
+    const el = new ES5ReflectReflect2();
+    assert.instanceOf(el, ES5ReflectReflect2);
+    assert.instanceOf(el, HTMLElement);
+    assert.equal(el.tagName, 'ES5-REFLECT-ELEMENT-2');
+  });
+
+  test('works with parser created elements', () => {
+    function ES5ReflectReflect3() {
+      return Reflect.construct(HTMLElement, [], this.constructor);
+    }
+    ES5ReflectReflect3.prototype = Object.create(HTMLElement.prototype);
+    ES5ReflectReflect3.prototype.constructor = ES5ReflectReflect3;
+
+    customElements.define('es5-reflect-element-3', ES5ReflectReflect3);
+
+    const container = document.createElement('div');
+    container.innerHTML = '<es5-reflect-element-3></es5-reflect-element-3>';
+    const el = container.querySelector('es5-reflect-element-3');
+    assert.instanceOf(el, ES5ReflectReflect3);
+    assert.instanceOf(el, HTMLElement);
+    assert.equal(el.tagName, 'ES5-REFLECT-ELEMENT-3');
+  });
+
+  test('reactions', () => {
+    function ES5ReflectElement4() {
+      return Reflect.construct(HTMLElement, [], this.constructor);
     }
+    ES5ReflectElement4.prototype = Object.create(HTMLElement.prototype);
+    ES5ReflectElement4.prototype.constructor = ES5ReflectElement4;
+    ES5ReflectElement4.observedAttributes = ['test'];
+    ES5ReflectElement4.prototype.connectedCallback = function() {
+      this.connectedCalled = true;
+    };
+    ES5ReflectElement4.prototype.disconnectedCallback = function() {
+      this.disconnectedCalled = true;
+    };
+    ES5ReflectElement4.prototype.attributeChangedCallback = function(name, oldValue, newValue) {
+      this.attributeChangedCalled = `${name}, ${oldValue}, ${newValue}`;
+    };
+    ES5ReflectElement4.prototype.adoptedCallback = function() {
+      this.adoptedCalled = true;
+    };
+
+    customElements.define('es5-reflect-element-4', ES5ReflectElement4);
+
+    const container = document.createElement('div');
+    document.body.appendChild(container);
+
+    const el = new ES5ReflectElement4();
+
+    container.appendChild(el);
+    assert(el.connectedCalled);
+
+    container.removeChild(el);
+    assert(el.disconnectedCalled);
+
+    el.setAttribute('test', 'good');
+    assert.equal(el.attributeChangedCalled, 'test, null, good');
+
+    const doc = document.implementation.createHTMLDocument();
+    doc.adoptNode(el);
+    assert(el.adoptedCalled);
+
+  });
+});
+
+suite('Native Shim with ES6 classes', () => {
+
+  // Do nothing if the browser does not natively support custom elements.
+  if (!window.customElements) {
+    test.skip('The native shim does not apply to this browser.', () => {});
+    return;
+  }
+
+  function createHTMLElementSubclassNamed(name) {
+    return eval('(function() { class ' + name + ' extends HTMLElement { constructor() { super(); }}; return ' + name + '; })')();
```

Comment:
Given that this entire file is about testing the shim, which only matters if you have a native implementation of custom elements, maybe we should just go with ES6 here and switch at [the point that it's passed to WCT](https://github.com/webcomponents/custom-elements/blob/adcbc4f1ce75832191b03b211185029ea03b6fc0/tests/index.html#L33). Although, it is nice to see that tests are being skipped. Or, is this because polyserve (via WCT) is trying to compile this class out? I need to double check.

Reply:
`polymer test` will compile out these classes if the browser it's testing on doesn't support them. However, if ES2015 isn't supported, then neither are native custom elements, so the checks in `suiteSetup` will skip the tests anyway. So, I've converted these to classes without using `eval`.